Src/IronPython/Runtime/ConversionWrappers.cs

```diff
@@ -5,18 +5,28 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Numerics;
 
 namespace IronPython.Runtime {
 
     public class ListGenericWrapper<T> : IList<T> {
         private readonly IList<object> _value;
+        // PEP 237: int/long unification (GH #52)
+        private static readonly bool IsBigIntWrapper = typeof(T) == typeof(BigInteger) || typeof(T) == typeof(BigInteger?);
 
         public ListGenericWrapper(IList<object> value) { _value = value; }
 
         #region IList<T> Members
 
         public int IndexOf(T item) {
-            return _value.IndexOf(item);
+            int pos = _value.IndexOf(item);
+            if (IsBigIntWrapper && item is BigInteger bi && bi >= int.MinValue && bi <= int.MaxValue) {
```

Comment:
I'm not sure it will get inlined or even compiled in an optimal way. For instance, even if `T` is `BigInteger`, going from `T item` to `BigInteger bi` likely involves boxing, similarly (in other places) casting a `BigInteger` result to `T`. I have put on my follow-up list to look into the prefromance of the conversion wrappers, but right now, short of creating dedicated classes for `BigInteger` and `Nullable<BigInteger>` or dropping down to IL, I don't have any ideas.